tests/components/SelectAutocompleteWithFetch.test.tsx

```diff
@@ -0,0 +1,258 @@
+import { act, fireEvent, render, screen, within } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+
+import SelectAutocompleteWithFetch from 'components/ui/Form/SelectAutocompleteWithFetch'
+
+type SearchedEntity = { idField: string; valueField: string }
+
+describe('SelectAutocompleteWithFetch', () => {
+  let fetchFunction: () => Promise<SearchedEntity[]>
+  let onUpdateSelected: (value: {
+    selected?: SearchedEntity
+    hasError: boolean
+  }) => void
+
+  const getOptions = () => screen.getByRole('listbox', { hidden: true })
+  const getSelect = () =>
+    screen.getByRole('combobox', { name: 'Label autocomplete' })
+  const callInitComponent = 1
+
+  beforeEach(() => {
+    // Given
+    fetchFunction = jest.fn(async () => [
+      { idField: 'valeur', valueField: 'valeur' },
+      { idField: 'valeur-avec-tiret', valueField: 'valeur-avec-tiret' },
+      { idField: 'valeur avec accent è', valueField: 'valeur avec accent è' },
+      {
+        idField: "valeur avec l'apostrophe",
+        valueField: "valeur avec l'apostrophe",
+      },
+    ])
+    onUpdateSelected = jest.fn()
+  })
+
+  describe('render', () => {
+    beforeEach(() => {
+      render(
+        <>
+          <label htmlFor='component'>Label autocomplete</label>
+          <SelectAutocompleteWithFetch
+            id='component'
+            fetch={fetchFunction}
+            fieldNames={{ id: 'idField', value: 'valueField' }}
+            onUpdateSelected={onUpdateSelected}
+            errorMessage='Error message'
+          />
+        </>
+      )
+    })
+
+    it("contient un getSelect avec une liste d'autocompletion", () => {
```

Comment:
titre pas très compréhensible. 
```suggestion
    it("contient un liste déroulante", () => {
```
le getSelect fait un getByRole d'un combobox